File: src/game/events/KilledEnemyEvent.java
package game.events;

import game.Bomberman;

/**
 * Fires a game event using Strategy Pattern;
 */
public class KilledEnemyEvent implements GameEvent {
    @Override
    public void invoke(Object arg) {
        Bomberman.getMatch().getCurrentLevel().onKilledEnemy();
    }
}

File: src/game/engine/events/game/KilledEnemyEvent.kt
package game.engine.events.game

import game.Bomberman
import game.engine.events.models.GameEvent


class KilledEnemyEvent : GameEvent {
    override fun invoke(arg: Any?) {
        Bomberman.getMatch().currentLevel!!.eventHandler.onKilledEnemy()
    }
}